### PR TITLE
fix rat and checkstyle issue

### LINKED ISSUE #15124: Druid build fails with git command '-no-show-signature'
Building Druid with the below command on master/v27.0 branches:
```
mvn clean package  install -DskipTests   -Pdist -Prat
````

the build  failed with the below error.

```
[ERROR] Failed to execute goal pl.project13.maven:git-commit-id-plugin:4.9.10:revision (default) on project druid: Could not complete Mojo execution...: pl.project13.core.NativeGitProvider$NativeCommandException: Git command exited with invalid status [128]: directory: `/root/bassa/apache/druid`, command: `git log -1 --pretty=format:%ct --no-show-signature HEAD`, stdout: ``, stderr: `fatal: unrecognized argument: --no-show-signature`
```

Solution:
comment the plugin from https://github.com/apache/druid/blob/master/pom.xml#L1569 to https://github.com/apache/druid/blob/master/pom.xml#L1607

Ex: 
```
            <!--plugin>
                <groupId>pl.project13.maven</groupId>
                <artifactId>git-commit-id-plugin</artifactId>
                <version>4.9.10</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>revision</goal>
                        </goals>
                    </execution>
                </executions>
                <configuration>
                    <dotGitDirectory>${project.basedir}/.git</dotGitDirectory>
                    <dateFormatTimeZone>Etc/UTC</dateFormatTimeZone>
                    <skipPoms>false</skipPoms>
                    <format>json</format>
                    <generateGitPropertiesFile>true</generateGitPropertiesFile>
                    <generateGitPropertiesFilename>${project.build.directory}/git.version</generateGitPropertiesFilename>
                    <failOnNoGitDirectory>false</failOnNoGitDirectory>
                    <useNativeGit>true</useNativeGit>
                    <excludeProperties>
                        <excludeProperty>git.build.user.email</excludeProperty>
                        <excludeProperty>git.build.host</excludeProperty>
                        <excludeProperty>git.commit.id.describe-short</excludeProperty>
                        <excludeProperty>git.commit.user.*</excludeProperty>
                        <excludeProperty>git.commit.message.*</excludeProperty>
                        <excludeProperty>git.closest.tag.*</excludeProperty>
                        <excludeProperty>git.commit.id.abbrev</excludeProperty>
                        <excludeProperty>git.dirty</excludeProperty>
                    </excludeProperties>
                    <gitDescribe>
                        <skip>false</skip>
                        <always>true</always>
                        <abbrev>7</abbrev>
                        <dirty>-dirty</dirty>
                        <forceLongFormat>true</forceLongFormat>
                    </gitDescribe>
                </configuration>
            </plugin-->

```

After commenting the above line, build crosses this error stage.

### PR DESCRIPTION
**Pull request by AlbericByte**
Fixes #15124

### Description

Building Druid with the below command on master/v27.0 branches:

mvn clean package  install -DskipTests   -Pdist -Prat
the build failed with the below error.

Update the following configuration to fix this issue:
+ checkstyle configurations
+ rat excludes configurations

#### Release note

<hr>

##### Key changed/added classes in this PR
 * `checkstyle-suppressions.xml`
 * `pom.xml`

<hr>

This PR has:

- [x] been self-reviewed.
- [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by FrankChen021** [comment]
The original issue is not re-produced on the master branch, but got a lots of checkstyle problems when running the command 

```
mvn package install -DskipTests -Pdist -Prat
```

The checkstyle problems are all for files in the generated-source directory of  `druid-sql` module. I think the change in this PR makes sense. 

But I would like to suggest **using global match to exclude all checks for the generated source** files like the following instead of adding supression rules one by one to the supression file.

```
<suppress files="[\\/]target[\\/]generated-test-sources[\\/]" checks=".*"/>
<suppress files="[\\/]target[\\/]generated-sources[\\/]" checks=".*"/>
```

File: codestyle/checkstyle-suppressions.xml [hunk 1]
```diff
@@ -29,23 +29,50 @@
   <!-- Code copied from TestNG to apply a bugfix -->
   <suppress checks="AvoidStaticImport" files="[\\/]org[\\/]testng[\\/]" />
 
+  <suppress checks="LocalFinalVariableName" files="[\\/]target[\\/]generated-sources[\\/]" />
```
Comment (FrankChen021) [inline codestyle/checkstyle-suppressions.xml:32]: Remove all rules from this file for generated-source and generated-test-sources and add the following two rules:

  <suppress checks=".*" files="[\\/]target[\\/]generated-test-sources[\\/]" />
  <suppress checks=".*" files="[\\/]target[\\/]generated-sources[\\/]" />

Comment (FrankChen021) [reply codestyle/checkstyle-suppressions.xml:32]: The URL in Line 27 of this file is out of date. Please also change it to: https://checkstyle.org/filters/suppressionfilter.html

Comment (AlbericByte) [reply codestyle/checkstyle-suppressions.xml]: good catch, fixed, Thanks

**Review by FrankChen021 — APPROVED** [review]

**Comment by FrankChen021** [comment]
This problem is only re-produced when running command 'mvn clean package install'. 
The checkstyle runs twice for one module, one is the at the very beginning of the build phase, that is the validate phase, the other is after the 'install' phase. 

I don't know why this happens. We can investigate this separately. For GHA, I think there's no need to make changes now until we find out the root cause why the checkstyle runs twice under the 'package install' command

File: pom.xml [hunk 2]
```diff
@@ -2108,9 +2108,14 @@
                                 <exclude>**/test/resources/**/*</exclude> <!-- test data for "old" ITs. -->
                                 <exclude>**/docker/client_tls/*</exclude> <!-- generated files for "old" ITs. -->
                                 <exclude>resources/data/**/*</exclude> <!-- test data for "new" ITs. -->
-                                <exclude>**/derby.log</exclude>
                                 <exclude>**/jvm.config</exclude>
                                 <exclude>**/*.avsc</exclude>
+                                <exclude>**/*.png</exclude>
```
Comment (cryptoe) [inline pom.xml:2113]: Why is this change required ?
